system/src/system_cloud_internal.cpp

```diff
@@ -1293,6 +1295,7 @@ int Spark_Handshake(bool presence_announce)
         LOG(INFO,"cloud connected from existing session.");
 
         publishSafeModeEventIfNeeded();
+        publishResetReasonIfNeeded();
         Send_Firmware_Update_Flags();
```

Comment:
feels like we should be able to move these out of the if/else block since theyre in both. I dont think their order particularly matters

Reply:
I would think that it is safe to do so.

Reply:
I agree. See my comments above.